includes/wikia/services/DataMartService.class.php

```diff
@@ -51,20 +51,30 @@ protected static function getPageviews ( $periodId, $startDate, $endDate = null,
 			}
 		}
 
-		$db = DataMartService::getDB();
-		$pageViews = ( new WikiaSQL() )->skipIf( self::isDisabled() )
-			->cacheGlobal( self::TTL )
-			->SELECT( "date_format(time_id,'%Y-%m-%d')" )->AS_( 'date' )
-				->FIELD( 'pageviews' )->AS_( 'cnt' )
-			->FROM( 'rollup_wiki_pageviews' )
-			->WHERE( 'period_id' )->EQUAL_TO( $periodId )
-				->AND_( 'wiki_id' )->EQUAL_TO( $wikiId )
-				->AND_( 'time_id' )->BETWEEN( $startDate, $endDate )
-			->runLoop( $db, function ( &$pageViews, $row ) {
-				$pageViews[$row->date] = $row->cnt;
-			} );
+		try {
+			$db = DataMartService::getDB();
+			$pageViews =
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$pageViews". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)